test/e2e/initializingworkspaces_test.go

```diff
@@ -0,0 +1,253 @@
+		Expect(err).NotTo(HaveOccurred())
+
+		lc2 := &kcpcorev1alpha1.LogicalCluster{}
+		err = cli.Cluster(ws2Path).Get(ctx, client.ObjectKey{Name: "cluster"}, lc2)
+		Expect(err).NotTo(HaveOccurred())
+		envtest.Eventually(GinkgoT(), func() (bool, string) {
+			return slices.Contains(lc1.Status.Initializers, kcpcorev1alpha1.LogicalClusterInitializer(initName)) && slices.Contains(lc2.Status.Initializers, kcpcorev1alpha1.LogicalClusterInitializer(initName)),
+				fmt.Sprintf("Initializer not set: %v", lc1.Status) + " " + fmt.Sprintf("%v", lc2.Status)
+		}, wait.ForeverTestTimeout, time.Millisecond*100, "failed to see initializers in both clusters")
+	})
+
+	Describe("with a multicluster provider and manager", func() {
+		var (
+			engaged             = sets.NewString()
+			p                   *initializingworkspaces.Provider
+			g                   *errgroup.Group
+			cancelGroup         context.CancelFunc
+			initializersRemoved = sets.NewString()
+		)
+
+		BeforeAll(func() {
+			cli, err := clusterclient.New(kcpConfig, client.Options{})
+			Expect(err).NotTo(HaveOccurred())
+			By("creating a multicluster provider for initializing workspaces")
+
+			// Get the initializing workspaces virtual workspace URL
+			wt := &tenancyv1alpha1.WorkspaceType{}
+			err = cli.Cluster(core.RootCluster.Path()).Get(ctx, client.ObjectKey{Name: workspaceTypeName}, wt)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(wt.Status.VirtualWorkspaces).ToNot(BeEmpty())
+
+			vwConfig := rest.CopyConfig(kcpConfig)
+			vwConfig.Host = wt.Status.VirtualWorkspaces[0].URL
+
+			// Create the provider with the initializer name from the WorkspaceType
+			p, err = initializingworkspaces.New(vwConfig, initializingworkspaces.Options{
+				InitializerName: initName,
+			})
+			Expect(err).NotTo(HaveOccurred())
+
+			By("creating a manager that uses the provider")
+			mgr, err = mcmanager.New(vwConfig, p, mcmanager.Options{})
+			Expect(err).NotTo(HaveOccurred())
+
+			By("creating a reconciler for LogicalClusters")
+			err = mcbuilder.ControllerManagedBy(mgr).
+				Named("kcp-initializer-controller").
+				For(&kcpcorev1alpha1.LogicalCluster{}).
+				Complete(mcreconcile.Func(
+					func(ctx context.Context, request mcreconcile.Request) (ctrl.Result, error) {
+						By(fmt.Sprintf("reconciling LogicalCluster %s in cluster %q", request.Name, request.ClusterName))
+						cl, err := mgr.GetCluster(ctx, request.ClusterName)
+						if err != nil {
+							return reconcile.Result{}, err
+						}
+						clusterClient := cl.GetClient()
+						lc := &kcpcorev1alpha1.LogicalCluster{}
+						list := &kcpcorev1alpha1.LogicalClusterList{}
+						if err := clusterClient.List(ctx, list); err != nil {
+							return reconcile.Result{}, fmt.Errorf("failed to list LogicalClusters: %w", err)
+						}
+
+						for _, item := range list.Items {
+							clusterName := logicalcluster.From(&item)
+							if string(clusterName) == request.ClusterName {
+								lc = &item
+								break
+							}
+						}
```

Comment:
Why this complicated dance to get the LogicalCluster? Can't you do a `Get` on the `cluster` name via the client?

Reply:
Wanted to leave the LIST in the test, as we are calling GET in the example.
We can remove it, I don't mind